service/file_transaction_service.py

```diff
@@ -16,18 +16,30 @@ def GetStatus(self, request, context):
         """Return the status of the service"""
         logger.request_service("GetStatus")
 
-        if self.__server.get_state() != server.State.AVAILABLE:
-            context.set_code = grpc.StatusCode.UNAVAILABLE.value[0]
-            context.set_details = grpc.StatusCode.UNAVAILABLE.name
+        # Lock service state for reading
+        with self.__server.get_state_locker().get_lock().gen_rlock():
+            logger.info("Service State:", str(self.__server.get_state_locker().get_current_service_state()))
+
+            if self.__server.get_state_locker().get_current_service_state() == server.State.AVAILABLE:
+                context.set_code = grpc.StatusCode.OK.value[0]
+                context.set_details = grpc.StatusCode.OK.name
+
+            if self.__server.get_state_locker().get_current_service_state() == server.State.UNAVAILABLE:
+                context.set_code = grpc.StatusCode.UNAVAILABLE.value[0]
+                context.set_details = grpc.StatusCode.UNAVAILABLE.name
+
+            # Check connection to Azure Blob Storage
+            try:
+                utility.block_blob_service.get_blob_service_properties()
+            except:
+                logger.exception("failed to connect to Azure Blob Storage")
```

Comment:
We need to log the stack trace.
Maybe do the following?
```
except Exception as error:
    logger.exception(error)
```

It looks like `logging` library implements that already.
https://stackoverflow.com/questions/4564559/get-exception-description-and-stack-trace-which-caused-an-exception-all-as-a-st

https://docs.python.org/3/library/logging.html#logging.Logger.exception
```
exception(msg, *args, **kwargs)
Logs a message with level ERROR on this logger. The arguments are interpreted as for debug(). Exception info is added to the logging message. This method should only be called from an exception handler.
```

Reply:
The current implementation does in fact log the stack trace, although it may not be clear in the code.